### PR TITLE
Replace "absolute()" with "resolve()" in pathlib objects

### PR DESCRIPTION
**Pull request by potiuk**
TIL that absolute() is an undocumented in Pathlib and that we
should use resolve() instead.

So this is it.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/batch_waiters.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def default_config(self) -> Dict:
         :rtype: Dict
         """
         if self._default_config is None:
-            config_path = Path(__file__).with_name("batch_waiters.json").absolute()
+            config_path = Path(__file__).with_name("batch_waiters.json").resolve()
             with open(config_path) as config_file:
                 self._default_config = json.load(config_file)
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/batch_waiters.py:125]: We don’t even need to resolve here, this should just work:

```python
with Path(__file__).with_name("batch_waiters.json").open() as config_file:
    self._default_config = json.load(config_file)
```

File: scripts/ci/pre_commit/pre_commit_check_pre_commit_hooks.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import sys
 from pathlib import Path
 
-sys.path.insert(0, str(Path(__file__).parent.absolute()))  # make sure common_precommit_utils is imported
+sys.path.insert(0, str(Path(__file__).parent.resolve()))  # make sure common_precommit_utils is imported
```
Comment (uranusjr) [inline scripts/ci/pre_commit/pre_commit_check_pre_commit_hooks.py:28]: Same here, `sys.path` accepts a relative path just fine and I don’t think we need to convert. (There are several places using this pattern and this should apply to them all.)

File: tests/utils/test_db_cleanup.py [hunk 3]
```diff
@@ -208,7 +208,7 @@ def test_no_models_missing(self):
         """
         import pkgutil
 
-        proj_root = Path(__file__).parents[2].absolute()
+        proj_root = Path(__file__).parents[2].resolve()
```
Comment (uranusjr) [inline tests/utils/test_db_cleanup.py:211]: Likely the same here as well.

File: airflow/providers/amazon/aws/hooks/batch_waiters.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/amazon/aws/hooks/batch_waiters.py:125]: We do not need to, but I prefer to do it in all cases where I am running any kind of path calculation that I am going to use later. The reason is that if I decide at a later point to use the path further and run it trough any command line tool, I do not have to worry whether it is absolute or not.

File: scripts/ci/pre_commit/pre_commit_check_pre_commit_hooks.py (hunk 2 above)
Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_pre_commit_hooks.py:28]: Same as above. It does not hurt. Call it being overcautiious but I do not want to exercise any extra mental power to remember where I can use relative path and where it is not safe. Using absolute path is always safe.

File: tests/utils/test_db_cleanup.py (hunk 3 above)
Comment (potiuk) [reply tests/utils/test_db_cleanup.py:211]: Same here :).

**Review by uranusjr — APPROVED** [review]

**Pull request merged**